panel/widgets/file_selector.py

```diff
@@ -280,20 +445,209 @@ def _select(self, event: param.parameterized.Event):
-        self._position -= 1
-        self._directory.value = self._stack[self._position]
-        self._update_files()
-        self._forward.disabled = False
-        if self._position == 0:
-            self._back.disabled = True
 
-    def _go_forward(self, event: param.parameterized.Event):
-        self._position += 1
-        self._directory.value = self._stack[self._position]
-        self._update_files()
+class FileTree(BaseFileSelector, _TreeBase):
+    """
+    FileTree renders a path or directory.
+    """
 
-    def _go_up(self, event: Optional[param.parameterized.Event] = None):
-        path = self._cwd.split(os.path.sep)
-        self._directory.value = os.path.sep.join(path[:-1]) or os.path.sep
-        self._update_files(True)
+    sort = param.Boolean(default=True, doc="""
+        Whether to sort nodes alphabetically.""")
+
+    max_depth = param.Integer(default=0, doc="""
+        The maximum depth to display in the tree.
+        If set to 0, all levels are shown.""")
+
+    _rename = {
+        'directory': None,
+        'file_pattern': None,
+        'refresh_period': None,
+        'root_directory': None,
+        'only_files': 'cascade',
+        'max_depth': None,
+        'value': 'checked',
+    }
+
+    def __init__(
+        self,
+        directory: AnyStr | os.PathLike | None = None,
+        fs: AbstractFileSystem | None = None,
+        **params,
+    ):
+        self._index = {}
+        super().__init__(directory, fs, **params)
+        self._internal_callbacks.append(
+            self.param.watch(self._set_data_from_directory, 'directory')
+        )
+        self._set_data_from_directory()
+
+    def _set_data_from_directory(self, event=None):
+        reset = False
+        if event and not (path:= fullpath(event.new)).startswith(self._root_directory):
+            self._root_directory = path
+            reset = True
+        try:
+            children, _ = self._get_children(
+                Path(self.directory).name, self.directory, depth=1
+            )
+        except FileNotFoundError:
+            self.directory = event.old
+            return
+        node_id = self._provider.normalize(self.directory)+self._provider.sep
+        self._nodes = [{
+            "id": node_id,
+            "text": Path(self.directory).name,
+            "icon": "jstree-folder",
+            "type": "folder",
+            "state": {"opened": True, "loaded": True},
+            "children": children
+        }]
+        self._reindex(reset=reset)
+        if reset:
+            self.value = []
+        if event:
+            self._param_change(event)
+
+    def _process_param_change(self, params):
+        if 'directory' in params:
+            params['nodes'] = self._nodes
+            del params['directory']
+        props = super()._process_param_change(params)
+        if 'value' in props:
+            props['checked'] = props.pop('value')
+        return props
```

Comment:
I think this section could be moved to the base tree, since it I think should always happen in any concrete implementation, and the generic Tree implementation needs it too.